jax/_src/pjit.py

```diff
@@ -1757,6 +1757,20 @@ def _resolve_wsc_args(axis_resources, shardings):
 # period is finished. The deprecation period expires 3 months from Feb 13, 2023.
 def with_sharding_constraint(x, shardings=UNSPECIFIED,
                              axis_resources=UNSPECIFIED):
+  """Mechanism to constrain the sharding of an Array.
```

Comment:
Maybe: `Mechanism to constrain the sharding of an Array inside a jitted computation`

Also add: `This is a strict constraint for the GSPMD partitioner and not a hint.`

Reply:
Done, thanks!